shared/actions/tracker.js

```diff
@@ -296,10 +316,34 @@ export function onFollow (username: string, localIgnore: bool): (dispatch: Dispa
   }
 }
 
-export function onClose (username: string): Action {
-  return {
-    type: Constants.onClose,
-    payload: {username}
+function _dismissWithToken (trackToken) {
+  const params : track_dismissWithToken_rpc = {
+    method: 'track.dismissWithToken',
+    param: {trackToken},
+    incomingCallMap: {},
+    callback: err => {
+      if (err) {
+        console.log('err dismissWithToken', err)
+      }
+    }
+  }
+  engine.rpc(params)
+}
+
+export function onClose (username: string): TrackerActionCreator {
+  return (dispatch, getState) => {
+    const trackToken = _getTrackToken(getState, username)
+
+    if (trackToken) {
```

Comment:
The reducer does in onclose which is called from multiple places

On Tuesday, May 24, 2016, Marco Munizaga <notifications@github.com> wrote:

> In shared/actions/tracker.js
> <https://github.com/keybase/client/pull/2873#discussion_r64501208>:
>
> > +    param: {trackToken},
> > +    incomingCallMap: {},
> > +    callback: err => {
> > +      if (err) {
> > +        console.log('err dismissWithToken', err)
> > +      }
> > +    }
> > +  }
> > +  engine.rpc(params)
> > +}
> > +
> > +export function onClose (username: string): TrackerActionCreator {
> > +  return (dispatch, getState) => {
> > +    const trackToken = _getTrackToken(getState, username)
> > +
> > +    if (trackToken) {
>
> What do you mean bookkeep? how is it doing that? needTrackTokenDismiss is
> never set to true, did you mean to do that here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/2873/files/7e8038253c9273f8d167ae13e4b46aed791d4103#r64501208>
>